lib/common/api/deprecate.js

```diff
@@ -18,101 +34,49 @@ const deprecate = {
       return console.warn(`(electron) ${message}`)
     }
   },
-  renameFunction: function (fn, oldName, newName) {
-    let warned = false
-    return function () {
-      if (!(warned || process.noDeprecation)) {
-        warned = true
-        deprecate.warn(oldName, newName)
-      }
-      return fn.apply(this, arguments)
-    }
-  },
-  removeFunction: (oldName) => {
-    if (!process.noDeprecation) {
-      deprecate.log(`The '${oldName}' function has been deprecated and marked for removal.`)
-    }
-  },
-  alias: function (object, oldName, newName) {
-    let warned = false
-    const newFn = function () {
-      if (!(warned || process.noDeprecation)) {
-        warned = true
-        deprecate.warn(oldName, newName)
-      }
-      return this[newName].apply(this, arguments)
-    }
-    if (typeof object === 'function') {
-      object.prototype[newName] = newFn
-    } else {
-      object[oldName] = newFn
-    }
-  },
+
   event: (emitter, oldName, newName) => {
-    let warned = false
+    const warn = newName.startsWith('-') /* internal event */
+      ? warnOnce(`${oldName} event`)
+      : warnOnce(`${oldName} event`, `${newName} event`)
     return emitter.on(newName, function (...args) {
-      if (this.listenerCount(oldName) === 0) return
-      if (warned || process.noDeprecation) return
-
-      warned = true
-      if (newName.startsWith('-')) {
-        deprecate.log(`'${oldName}' event has been deprecated.`)
-      } else {
-        deprecate.warn(`'${oldName}' event`, `'${newName}' event`)
+      if (this.listenerCount(oldName) !== 0) {
+        warn()
+        this.emit(oldName, ...args)
       }
-      this.emit(oldName, ...args)
     })
   },
-  removeProperty: (object, deprecated, ignoreMissingProps = false) => {
-    let warned = false
-    let warn = () => {
-      if (!(warned || process.noDeprecation)) {
-        warned = true
-        deprecate.log(`The '${deprecated}' property has been deprecated and marked for removal.`)
-      }
-    }
 
-    if (!(deprecated in object)) {
-      if (ignoreMissingProps) return
-      throw new Error('Cannot deprecate a property on an object which does not have that property')
-    }
+  removeProperty: (o, removedName) => {
+    const warn = warnOnce(removedName)
 
-    let temp = object[deprecated]
-    return Object.defineProperty(object, deprecated, {
+    // if the property's already been removed, warn about it
+    if (!(removedName in o)) warn()
```

Comment:
see above, there is more than one reason that a property might not exist on an object and i think it may be important to differentiate between "it was already removed" and "it may or may not have been there to begin with"

Reply:
I'm open to that idea. When is that important?

Reply:
Added a clearer warning in https://github.com/electron/electron/pull/14579/commits/deeb59de683146f249423a2528241c31a443d9af